From 194ef16dc324cde02a50a9a3ecea7189bf86dd3d Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Sat, 23 May 2020 19:21:46 +0300
Subject: [PATCH] debc: Don't use IPC::Run for piping from dpkg-deb

IPC::Run pipes hang for >64k of data, and a regular open is simpler
anyway.

See https://github.com/toddr/IPC-Run/issues/126
---
 scripts/debi.pl | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/scripts/debi.pl b/scripts/debi.pl
index c833b1fc..d8eb824b 100755
--- a/scripts/debi.pl
+++ b/scripts/debi.pl
@@ -32,8 +32,6 @@ use filetest 'access';
 use Cwd;
 use Dpkg::Control;
 use Dpkg::Changelog::Parse qw(changelog_parse);
-use IPC::Run qw(run);
-use IO::Handle;
 use Dpkg::IPC;
 
 my $progname = basename($0, '.pl');    # the '.pl' is for when we're debugging
@@ -373,8 +371,7 @@ for (split(/\n/, $ctrl->{Files})) {
     if (   (($progname eq 'debi') && (/ (\S*\.deb)$/))
         || (($progname eq 'debc') && (/ (\S*\.u?deb)$/))) {
         my $deb    = $1;
-        my $stdout = IO::Handle->new();
-        run(['dpkg-deb', '-f', $deb], '>pipe', \*$stdout);
+        open(my $stdout, '-|', 'dpkg-deb', '-f', $deb);
         my $fields = Dpkg::Control->new(name => $deb, type => CTRL_PKG_DEB);
         $fields->parse($stdout, $deb);
         my $pkg = $fields->{Package};
-- 
2.26.1

